src/common/com/intellij/plugins/haxe/model/HaxeClassModel.java

```diff
@@ -102,9 +102,17 @@ public boolean isAbstract() {
   public HaxeTypeOrAnonymous getAbstractUnderlyingType() {
     if (!isAbstract()) return null;
     PsiElement[] children = getPsi().getChildren();
-    if (children.length >= 2) {
-      if (children[1] instanceof HaxeTypeOrAnonymous) {
-        return (HaxeTypeOrAnonymous)children[1];
+    // FIX: support list of metas before ComponentName
+    for(int i = 0; i < children.length; ++i) {
+      PsiElement child = children[i];
+      if(child instanceof HaxeComponentName) {
+        if(i + 1 < children.length) {
+          child = children[i + 1];
+          if(child instanceof HaxeTypeOrAnonymous) {
+            return (HaxeTypeOrAnonymous)child;
+          }
```

Comment:
For future reference, there is a UsefulPsiTreeUtil.getChildOfType function, which you can use to search for a specific element type.  That way you don't have to write the loops yourself.